backend/migrations/000004_add_metadata_to_thread.up.sql

```diff
@@ -0,0 +1,9 @@
+ALTER TABLE thread
+ADD COLUMN metadata JSON;
```

Comment:
Suggestion: Should we make this a `JSONB` type? The current implementation updates the `metadata` column transactionally in the application layer, but there may be use cases in the future where we'd want to push updating/querying to the database. For example, filtering threads by assistant type.

This could potentially impact the way the `metadata` column is updated.

Reply:
Good idea. Adjusted.